packages/starknet-snap/src/rpcs/estimateFee.ts

```diff
@@ -0,0 +1,151 @@
+import { TransactionType, type Invocations } from 'starknet';
+import type { Infer } from 'superstruct';
+import { object, string, assign, boolean } from 'superstruct';
+
+import {
+  AddressStruct,
+  AuthorizableStruct,
+  BaseRequestStruct,
+  AccountRpcController,
+  toJson,
+  logger,
+} from '../utils';
+import { ACCOUNT_CLASS_HASH } from '../utils/constants';
+import {
+  addFeesFromAllTransactions,
+  estimateFeeBulk,
+  getAccContractAddressAndCallData,
+  isAccountDeployed,
+} from '../utils/starknetUtils';
+
+export const EstimateFeeRequestStruct = assign(
+  object({
+    address: AddressStruct,
+    contractAddress: AddressStruct,
+    contractFuncName: string(),
+    contractCallData: string(),
```

Comment:
i think may be better to follow the same interface as starknet.js?
```suggestion
    Invocations: array(InvocationStruct),
    details: optional(UniversalDetailsStruct)
```

Reply:
I agree, but can we do this uniformisation of fee estimation handling in later PR ? As this PR is to be able to revamp sendTransaction.